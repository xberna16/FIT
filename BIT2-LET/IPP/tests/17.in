<?xml version="1.0" encoding="UTF-8"?>
<program language="IPPcode18"><instruction order="1" opcode="DEFVAR"><arg1 type="var">GF@A</arg1></instruction><instruction order="2" opcode="DEFVAR"><arg1 type="var">GF@B</arg1></instruction><instruction order="3" opcode="DEFVAR"><arg1 type="var">GF@C</arg1></instruction><instruction order="4" opcode="DEFVAR"><arg1 type="var">GF@Vysledek</arg1></instruction><instruction order="5" opcode="MOVE"><arg1 type="var">GF@A</arg1><arg2 type="string">ABCD</arg2></instruction><instruction order="6" opcode="MOVE"><arg1 type="var">GF@B</arg1><arg2 type="int">0</arg2></instruction><instruction order="7" opcode="STRLEN"><arg1 type="var">GF@C</arg1><arg2 type="var">GF@A</arg2></instruction><instruction order="8" opcode="WRITE"><arg1 type="var">GF@A</arg1></instruction><instruction order="9" opcode="LABEL"><arg1 type="label">while</arg1></instruction><instruction order="10" opcode="JUMPIFEQ"><arg1 type="label">end</arg1><arg2 type="var">GF@B</arg2><arg3 type="var">GF@C</arg3></instruction><instruction order="11" opcode="GETCHAR"><arg1 type="var">GF@Vysledek</arg1><arg2 type="var">GF@A</arg2><arg3 type="var">GF@B</arg3></instruction><instruction order="12" opcode="WRITE"><arg1 type="var">GF@Vysledek</arg1></instruction><instruction order="13" opcode="SETCHAR"><arg1 type="var">GF@Vysledek</arg1><arg2 type="int">0</arg2><arg3 type="string">X</arg3></instruction><instruction order="14" opcode="WRITE"><arg1 type="var">GF@Vysledek</arg1></instruction><instruction order="15" opcode="ADD"><arg1 type="var">GF@B</arg1><arg2 type="var">GF@B</arg2><arg3 type="int">1</arg3></instruction><instruction order="16" opcode="JUMP"><arg1 type="label">while</arg1></instruction><instruction order="17" opcode="LABEL"><arg1 type="label">end</arg1></instruction></program>
