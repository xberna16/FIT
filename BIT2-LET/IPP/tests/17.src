.IPPcode18
DEFVAR GF@A
DEFVAR GF@B
DEFVAR GF@C
DEFVAR GF@Vysledek
MOVE GF@A string@ABCD
MOVE GF@B int@0
STRLEN GF@C GF@A
WRITE GF@A
LABEL while
JUMPIFEQ end GF@B GF@C
GETCHAR GF@Vysledek GF@A GF@B
WRITE GF@Vysledek
SETCHAR GF@Vysledek int@0 string@X
WRITE GF@Vysledek
ADD GF@B GF@B int@1
JUMP while
LABEL end
