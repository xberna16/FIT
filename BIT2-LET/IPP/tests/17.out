ABCD
A
X
B
X
C
X
D
X
